### PR TITLE
Add scalar charge observable

### PR DESCRIPTION
**Pull request by guilara**
## Proposed changes

Add functions and tags that observe the scalar charge for the ScalarTensor system

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/ScalarTensor/ScalarCharge.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+template <typename Frame>
+void ScalarTensor::scalar_charge_integrand(
+    const gsl::not_null<Scalar<DataVector>*> result,
+    const tnsr::i<DataVector, 3, Frame>& phi,
+    const tnsr::I<DataVector, 3, Frame>& unit_normal_vector,
+    const Scalar<DataVector>& area_element) {
+  set_number_of_grid_points(result, area_element);
+  for (auto& component : *result) {
+    component = 0.0;
+  }
```
Comment (nilsvu) [inline src/PointwiseFunctions/ScalarTensor/ScalarCharge.cpp]: you can delete these lines

File: src/PointwiseFunctions/ScalarTensor/ScalarCharge.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  // Project the scalar gradient on the normal vector
+  get(*result) = phi.get(0) * get<0>(unit_normal_vector);
+  for (size_t j = 1; j < 3; ++j) {
+    get(*result) += phi.get(j) * unit_normal_vector.get(j);
+  }
```
Comment (nilsvu) [inline src/PointwiseFunctions/ScalarTensor/ScalarCharge.cpp]: (optional) This is `tenex::evaluate(result, phi(ti::i) * unit_normal_vector(ti::I))` if you want to give tensor expressions a try :)

File: src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ * For a spherically symmetric scalar, this value will coincide with the value
+ * as extracted from the $r^{-1}$ decay in
+ * \f[ \Psi \sim \phi_\infty + q / r + \cdots~. \f]
+ */
+template <typename Frame>
+void scalar_charge_integrand(
+    const gsl::not_null<Scalar<DataVector>*> result,
+    const tnsr::i<DataVector, 3, Frame>& phi,
+    const tnsr::I<DataVector, 3, Frame>& unit_normal_vector,
+    const Scalar<DataVector>& area_element);
```
Comment (nilsvu) [inline src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp]: you can delete the `area_element` argument

File: src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp [hunk 4]
```diff
@@ -0,0 +1,90 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <utility>
+
+#include "ApparentHorizons/StrahlkorperGr.hpp"
```
Comment (nilsvu) [inline src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp]: I think you can delete some of these includes, to reduce the dependencies of the ScalarTensor library

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: src/PointwiseFunctions/ScalarTensor/ScalarCharge.cpp (hunk 1 above)
Comment (guilara) [reply src/PointwiseFunctions/ScalarTensor/ScalarCharge.cpp]: Deleted

File: src/PointwiseFunctions/ScalarTensor/ScalarCharge.cpp (hunk 2 above)
Comment (guilara) [reply src/PointwiseFunctions/ScalarTensor/ScalarCharge.cpp]: Okay, lets give it a try :)

File: src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp (hunk 3 above)
Comment (guilara) [reply src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp]: Removed

File: src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp (hunk 4 above)
Comment (guilara) [reply src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp]: Right. I reduced the includes.

**Comment by guilara** [comment]
And thanks @nilsvu for being so quick and thorough in reviewing these PRs :)

File: src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+ * \f{align*}
+ * q = - \dfrac{1}{4 \pi} \oint dA \Phi_i n^{i},
+ * \f}
+ * where \f$ n^{i} \f$ is the unit (outward) normal of the surface.
+ *
+ * For a spherically symmetric scalar, this value will coincide with the value
+ * as extracted from the $r^{-1}$ decay in
+ * \f[ \Psi \sim \phi_\infty + q / r + \cdots~. \f]
+ */
+template <typename Frame>
```
Comment (nilsvu) [inline src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp]: Will you need instantiations for the different frames? Else you can just remove the `Frame` parameter altogether. `tnsr::i` defaults to inertial frame.

File: src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+struct ScalarChargeIntegrandCompute : ScalarChargeIntegrand, db::ComputeTag {
+  static constexpr size_t Dim = 3;
+  using base = ScalarChargeIntegrand;
+  static constexpr auto function =
+      static_cast<void (*)(gsl::not_null<Scalar<DataVector>*>,
+                           const tnsr::i<DataVector, Dim, Frame>&,
+                           const tnsr::I<DataVector, Dim, Frame>&)>(
+          &ScalarTensor::scalar_charge_integrand<Frame>);
```
Comment (nilsvu) [inline src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp]: you don't even need the `static_cast` since you have only a single overload.

**Review by nilsvu — DISMISSED** [review]
Squash directly if you want to do these

File: src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp (hunk 6 above)
Comment (guilara) [reply src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp]: Okay, I have removed the `static_cast`

File: src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp (hunk 5 above)
Comment (guilara) [reply src/PointwiseFunctions/ScalarTensor/ScalarCharge.hpp]: Also removed the `Frame` parameter

File: src/PointwiseFunctions/ScalarTensor/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  GeneralRelativity
+  LinearAlgebra
+  Options
+  SphericalHarmonics
+  ApparentHorizons
+  PRIVATE
+  CoordinateMaps
+  Domain
+  INTERFACE
+  Observer
```
Comment (nilsvu) [inline src/PointwiseFunctions/ScalarTensor/CMakeLists.txt]: I think you can remove a bunch of these dependencies (sorry for not catching this earlier)

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: src/PointwiseFunctions/ScalarTensor/CMakeLists.txt (hunk 7 above)
Comment (guilara) [reply src/PointwiseFunctions/ScalarTensor/CMakeLists.txt]: No worries, I removed a few of them looking at other folders in `PointwiseFunctions`. The test still passes in my laptop. Is there a systematic way to choose which libraries are strictly necessary to link?

Comment (nilsvu) [reply src/PointwiseFunctions/ScalarTensor/CMakeLists.txt]: The way to determine which libraries to link is through your includes:

- Libraries you include _only_ in cpp files are `PRIVATE`
- Libraries you include _only_ in hpp files are `INTERFACE`
- Libraries you include in both cpp and hpp files are `PUBLIC` (note that this also includes libraries you include in cpp files through other headers in your library)

**Review by nilsvu — APPROVED** [review]

**Comment by nilsvu** [comment]
You can also fix the library dependencies in another PR

**Pull request merged**